### ISSUE TITLE
Docker auto build stuck forever

### ISSUE THREAD

**Issue by darren** [docker/hub-feedback#1935]
## Problem description

repository:  https://hub.docker.com/r/darrenhoo/centos5-go-bootstrap

Build log shows that it is finished, but its status is in `in progress`

<img width="400" alt="auto-build-hangs" src="https://user-images.githubusercontent.com/12817/71863466-e7caa080-3137-11ea-9105-0e8e93fa0b54.png">

**Comment by darren** [comment]
Finnally, it finished:

Created: January 7, 2020 02:37 AM
Finished: January 7, 2020 10:37 AM

Queue time: 240m
Build time: 240m

**Comment by meowsbits** [comment]
Bumping to reopen this. I'm experiencing this today, and

```
Queue time: 240m
Build time: 240m
```

looks pretty fishy.  Seems like a `build completed` hook somewhere just isn't working.

**Comment by tcosentino** [comment]
Seeing my build stuck today as well, 4 hours now. Also noticing duplicate builds when I push to github

**Comment by NotExpectedYet** [comment]
Same have this currently, build complete but still stuck holding up other pushes.

**Comment by NotExpectedYet** [comment]
Mine just finally completed like 4 hours after. 

It was probably my fault tbh I'd cancelled the build half way through building.

**Comment by tcosentino** [comment]
> Same have this currently, build complete but still stuck holding up other pushes.

Not sure if it is related to my issue, but somehow I had a second hook created in github and each build was triggering two times. I'm not 100% sure that was related to the build being hung up or not, but when I removed that it seemed to get better.

**Comment by purefan** [comment]
Currently seeing this in one of my builds, going on for 2 hours but it usually finishes in 15 minutes

**Comment by Simplatex** [comment]
IN PROGRESS forever !

![image](https://user-images.githubusercontent.com/64586809/80913902-73bda380-8d8b-11ea-87bd-c898d0a7acb3.png)

**Comment by machaven** [comment]
> > Same have this currently, build complete but still stuck holding up other pushes.
> 
> Not sure if it is related to my issue, but somehow I had a second hook created in github and each build was triggering two times. I'm not 100% sure that was related to the build being hung up or not, but when I removed that it seemed to get better.

I have removed duplicate webhooks some time ago, but i'mt still experience this issue. I had this issue on the 23rd April and again today (3rd June). According to the build logs the build finished, but it get's stuck in progress. Cancelling does not help. I sent an email to support on both occasions.

**Comment by NotExpectedYet** [comment]
Mine was stuck because I'd cancelled it at an in-opportune time in the build process I gather. 

I didn't have any second webhooks created anyway.

**Comment by johnvcoleman-w24** [comment]
Piling on. Build finished in 1 minute, build showed in progress for several hours after the build completed, so I hit cancel. Several hours later it's still trying to cancel the build. In short, a 1 minute build has been "running" for 4 hours now.

This issue comes back every few months... the rate of regression on this issue is horrible.

**Comment by purefan** [comment]
I imagine this isnt meant to be a proper deployment solution, I have been using it as such for a hobby project where Watchtower just updates my containers when a new version of the image is available, but perhaps we should just see it as "eventually available"

**Comment by ahpook** [comment]
Just got bit by a variant of this on https://hub.docker.com/r/relaysh/pulumi-step-run 
The build against 9c03e9e finished and pushed successfully according to the log, where the last message was "Build finished" but it was still in progress according to the dashboard. I clicked "Cancel" and now it's in "Cancelling" status and has been for several hours. Further builds are stacked up in "Pending" waiting for this cancellation.

**Comment by savager** [comment]
This is clearly an issue when you cancel a build that has already failed or completed. Not sure why this can't be targeted.

**Comment by arvtiwar** [comment]
I have the same issues, there is no way to clean

**Comment by ondrovic** [comment]
Experiencing this today, didn't cancel, just normal auto triggered build

**Comment by EthraZa** [comment]
Same here, 4 hours already

**Comment by machaven** [comment]
We had the same on Friday and today it's happening again.

**Comment by Madpeterz** [comment]
I have this issue, a C# dotnetcore build failed 
"DockerBuild -o /app/build' returned a non-zero code: 1" 
after that it build hanged.

**Comment by Madpeterz** [comment]
workaround: Delete the docker hub repo and recreate with a new name
side-effects: Anyone using the old repo name will be forced to update

**Comment by mdrive20** [comment]
same issue for me, happened twice over 2 weeks.  The last time I turned down my remote instance which pulls new images using watchdog, and I noticed it didn't take as long.  I wonder if something to do with remote images trying to pull in while its being built?  I did notice when I manually tried pulling in new image during time that docker repo was stalled, my remote machine did download the new image (took about an hour though for a 2 min. job normally).

**Comment by oblalex** [comment]
Same here: built and published in 1 minute, hanging for 2h and counting.

**Comment by pupattan** [comment]
Is there any way to  clean ?  I tries to stop that also hanged

**Comment by shawnaxsom** [comment]
Hi everyone, thanks for your feedback.

Along with the recent comments here, we have had a few support tickets come in recently regarding hanging builds. The builds may time out after four hours after becoming stuck. We fixed one cause in September, but I see there have been quite a few reports from everyone after that point.

I've created a sprint ticket on our end to investigate.

**Comment by dylankbuckley** [comment]
@shawnaxsom This is still very much an issue - do you have any updates?

**Comment by alexks02** [comment]
The same. Bump

**Comment by shawnaxsom** [comment]
Sorry all, it has been pushed to the current sprint. It's earmarked for one of our engineers to start on after he finishes up his current tasks this sprint, hopefully soon. I'm not sure how difficult of an issue it will be to reproduce and address, so I can't guarantee a timeline, but it is a priority for us.

It may have been a little worse today as processing events backed up slightly during one of our deployments. That engineer is helping get the processing sped up for today's processing build events. I'll share your comments here with him.

**Comment by shawnaxsom** [comment]
How often has this been happening lately?

We don't think there is a bug that is causing the issue. The comments we've seen in this thread have primarily been a queuing issue, where the queue of builds becomes backed up.

We've changed the services infrastructure in the past month, and we've been addressing some bottlenecks in our overall infrastructure that hopefully have lowered the number of times this has happened to you recently. Let me know otherwise if it is happening often (and at what frequency would help).

**Comment by dylankbuckley** [comment]
Every 2nd day.

**Comment by savager** [comment]
Can’t say I’ve had it recently because I avoid it like the plague, but for me it was easy to replicate: accidentally push a branch when it was already building. Triggering another build.

**Comment by ScottHarben** [comment]
@shawnaxsom - This has been the case for me all morning:

I created a new repository and connected the auto build to my GitHub repository and clicked 'Save and Build'. The Latest Build Status showed it was successful, but the Recent Build section showed that my initial build was queued. After about 30 minutes of the build being in 'queued' status, I pushed a change to my GitHub master branch to see if this would "unclog" the queue. Now the Recent Build section is showing that the initial build was successful, but the build triggered by the GitHub push has been stuck in 'In Progress' status for over an hour. The build logs show that the build has finished.

Thanks for the help

<img width="953" alt="Capture" src="https://user-images.githubusercontent.com/38980543/101933251-edcbcb80-3ba9-11eb-922c-8c0d4c51a029.PNG">

**Comment by mcarans** [comment]
@shawnaxsom I have a hanging build again and have had this several times previously. In the absence of a fix for the hanging, I think there needs to be a way to force kill a build as cancelling doesn't seem to help.

**Comment by mayinghan** [comment]
experiencing this now. im a paid customer

**Comment by MattRiddell** [comment]
Yeah, same here (paid customer) - noticed in some other threads people are recommending building on another service like azure (for free) then pushing to docker hub.

**Comment by f3l1x** [comment]
I am facing same issue for almost week. I am using Github + automated build.

**Comment by dylankbuckley** [comment]
We just moved over to DigitalOceans container registry. DockerHub is a joke, the company is probably going to go under soon.

**Comment by laygir** [comment]
@shawnaxsom there is clearly a bug somewhere that causes bunch of people lose tens of hours due to builds getting stuck, cancel buttons not working and without any way to abort the operation.

Has there been any actual progress on your end since your last message about 6 months ago? Or is this task just dumped to the backlog I wonder.

**Comment by mdrive20** [comment]
I talked to one of their product managers a few weeks ago, he said they are fully aware of the problem and this was on their schedule as the next thing to get fixed

**Comment by shawnaxsom** [comment]
Yes, we made a few smaller-scale adjustments that weren't enough. We've had more load on Autobuild this year as well. So we'll need to dive a little deeper.

We did identify next steps, and another team is slated to focus on this as the next work item from what I last heard. I'll check on where that stands.

**Comment by shawnaxsom** [comment]
Note we're investigating slowness in Autobuild today, we've narrowed it down to high CPU utilization in a message queue.

We don't have metrics history past 14 days to know if it's been a problem in the past, but this is at least the issue for any slowness you've seen today (it's been worse than normal).

It isn't resolved yet, we're investigating reasons for high CPU further.

**Comment by laygir** [comment]
Thank you for the updates 🙏 

Really looking forward to this being resolved soon. We are suffering from this issue about twice a week in recent months. And it always happens at that Murphy time. This week we had a hotfix production deploy and that hotfix build started 2 days ago and was still running last night. So it is quite impactful when there is such situation, stakeholders expecting a release and we are unable to explain why we are not able to deploy the fix. Then after waiting some time we end up building locally and so on..

Good luck with the findings and I hope we all have this tackled soon 🙌

**Comment by laygir** [comment]
Coming back here after 6 more months. Was there any progress on this? 👀

**Comment by thepixelmonk** [comment]
Happening to us again. Any progress on this?

**Comment by mdrive20** [comment]
Happening today again for us, currently waiting 2 hrs. Several previous builds from github auto-build triggers failed, current one is now hung up

**Comment by ingshtrom** [comment]
We had some issues yesterday surrounding our asynchronous workflows, like autobuilds, so that is likely why there were issues yesterday. The pending builds should have flushed through by now, but if not please cancel any pending builds and retry them from GitHub.

**Comment by laygir** [comment]
@ingshtrom thanks for your message, but that doesn't really provide an answer for the actual topic of this open issue.

**Comment by burnsjeremy** [comment]
I am experiencing the same issue, build hung up...logs show that it completed and passed but not the UI and a new image will not queue to build out new images. Really wish there was a way to terminate it when cancel d work.

**Comment by jwoodrow** [comment]
I'm experiencing the same issue I had a build stuck after the "Build finished" log and now every build with the same tag is stuck in pending (I've canceled it since it already did its job completely)

**Comment by senica** [comment]
Just want to acknowledge that I'm experiencing the same issue. This thread gives me hope that it will clear up by tomorrow :) Thankfully it's not a critical release.

**Comment by ErcinDedeoglu** [comment]
I'm experiencing the same thing. And it is sad to see it has been a bug for years...

**Comment by ViktorsLV** [comment]
Happening again..

**Comment by Ben-Epstein** [comment]
I opened a new issue because maybe this one got lost https://github.com/docker/hub-feedback/issues/2348
